src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import core.basesyntax.handler.BalanceOperationHandler;
+import core.basesyntax.handler.OperationHandler;
+import core.basesyntax.handler.PurchaseOperationHandler;
+import core.basesyntax.handler.ReturnOperationHandler;
+import core.basesyntax.handler.SupplyOperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.impl.ReaderServiceImpl;
+import core.basesyntax.service.impl.ReportServiceImpl;
+import core.basesyntax.service.impl.TransactionParserServiceImpl;
+import core.basesyntax.service.impl.WriterServiceImpl;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    private static final String INPUT_FILE_PATH = "data.csv";
+    private static final String OUTPUT_FILE_PATH = "report.csv";
+    private static Map<FruitTransaction.Operation, OperationHandler> handlerMap;
+
+    static {
+        handlerMap = new HashMap<>();
+        handlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        handlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
+        handlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        handlerMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperationHandler());
+    }
+
+    public static void main(String[] args) {
+        OperationStrategy operationStrategy = new OperationStrategyImpl(handlerMap);
+
+        List<String> data = new ReaderServiceImpl().readFromFile(INPUT_FILE_PATH);
+        List<FruitTransaction> transactions = new TransactionParserServiceImpl().parse(data);
+
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getByOperation(transaction.getOperation());
+            handler.apply(transaction);
+        }
```

Comment:
It’s  business logic so you should  move it from Main to some OperationProcessorService